server/fleet/users.go

```diff
@@ -403,6 +403,7 @@ func saltAndHashPassword(keySize int, plaintext string, cost int) (hashed []byte
 	if err != nil {
 		return nil, "", err
 	}
+	salt = salt[:keySize]
```

Comment:
not an issue for ASCII, but do we support unicode password characters?

Reply:
It looks like [we do want to support unicode characters](https://fleetdm.slack.com/archives/C01EZVBHFHU/p1706813742549099?thread_ts=1706813356.873879&cid=C01EZVBHFHU) – I'll think about this a bit more!

Reply:
Given that the unicode issue is Fleet-wide, I'm going to schedule a meeting and/or put in a feature fest topic to track it, making sure this specific area  gets included in considerations.

Reply:
@noahtalerman  the password length restriction is on bytes, not characters. A unicode character can potentially be more than 1 byte, meaning, the hard-coded error message currently surfaced is not accurate. A simple solution would be to update the error message to say "48 bytes" instead of the ambiguous "48." What do you think? cc @mostlikelee

Reply:
Hmm, I see what you're saying: "bytes" is more accurate.

I think "characters" is more user friendly than "bytes"

I think this is edge-case enough to prioritize the user friendly "characters" over accuracy.

@rachaelshaw, what do you think?

Also, do other apps allow unicode characters in passwords? Should we?

Reply:
@noahtalerman @jacobshandling yeah I don't think we should use "bytes" in user-facing messaging. We could potentially add something to the FAQ section of the docs explaining the more detailed password requirements, and link to that. (@noahtalerman lmk if you want to bring this through drafting.)

If there's a cap on password size, I think it's going to continue to be a headache to support unicode in passwords. (For example: type `"🧑🏻‍🎄".length` in the browser console and see the result 🤡). 

Maybe we could narrow down the allowed characters to be just the characters that password managers commonly use in their auto-generated passwords. [Here's an example](https://support.kaspersky.com/UIS_Account_Portal/en-US/196079.htm) I found of some clearly-documented password requirements in the wild, we could do something like that in the future.